hack/local-setup-mgc.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_SETUP_DIR="$(dirname "${BASH_SOURCE[0]}")"
+source "${LOCAL_SETUP_DIR}"/.binEnv
+source "${LOCAL_SETUP_DIR}"/.setupEnv
+source "${LOCAL_SETUP_DIR}"/.clusterUtils
+source "${LOCAL_SETUP_DIR}"/.argocdUtils
+source "${LOCAL_SETUP_DIR}"/.cleanupUtils
+source "${LOCAL_SETUP_DIR}"/.deployUtils
+
+INGRESS_NGINX_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/ingress-nginx
+METALLB_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/metallb
+CERT_MANAGER_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/cert-manager
+EXTERNAL_DNS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/external-dns
+ARGOCD_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/argocd
+ISTIO_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/istio/istio-operator.yaml
+GATEWAY_API_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/gateway-api
+REDIS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/kuadrant/redis
+THANOS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/thanos
+PROMETHEUS_FOR_FEDERATION_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/prometheus-for-federation
+GRAFANA_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/grafana
+
+set -e pipefail
+
+cleanupMGC
```

Comment:
It would be good if we can get to a point where this script was idempotent and this cleanup would not be required . What makes this required here currently? OLM?

Reply:
The cleanup also gives an option to "uninstall" our stuff. Like deleting Gateway and asking the user to run the reconciler so the DNS records will be cleaned up and nuking namespaces. The idea being that when I'm playing with stuff locally the `make local-setup-mgc` would be "bring me back to the point when everything was working". 
In the context of this script only OLM really needs to either be cleaned up or the install logic should be adjusted. What would be the benefits of not needing a cleanup before the installation?

Reply:
If someone really wanted to cleanup first they could just call the cleanup script, or make target, before calling this script.

Reason to make it cleanup would be to bring it inline with how any configuration management tooling would work i.e. ArgoCD, by just reapplying config to a known set of clusters. If for instance it fails half way through for some networking issues (Happens to me often), then i don't really need it to clean everything up before it applies stuff again. I just need to keep trying until it completes and brings my clusters into the expected state.

This is not something you need to change in this PR, just something to consider when looking at follow on tasks for enhancements to these scripts.

Reply:
created #413 to revisit this part